src/cci/cas_cci.c

```diff
@@ -712,14 +713,24 @@ cci_disconnect (int mapped_conn_id, T_CCI_ERROR * err_buf)
 
       get_last_error (con_handle, err_buf);
     }
-  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool (con_handle->id) >= 0)
+  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT])
     {
-      cci_end_tran_internal (con_handle, CCI_TRAN_ROLLBACK);
-      API_ELOG (con_handle, 0);
+      if (con_handle->id < 0)
```

Comment:
What about define 'int rv;' inside this else {} block. (trivial)

```
else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT])
  {
    int rv;
    ...
```

Reply:
Can I ask about the reason?
Because I searched the "rv" variable which is used in CUBRID, and then almost they are not used intermediate in the code.

So, I think It should be the reason to move the variable'rv'

Reply:
This protection code works because I set the con_handle->id as -1 and then core dump occured.